### ISSUE TITLE
Editing the contents of a multiple receiver in a dialogue

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4069]
In a multiple receiver I can select one or more variables and then delete them.  But I would often prefer to replace them by new variables from the selector.  I don't think this is possible now?  (In a single receiver it replaces nicely of course.)

When I move to the data selector, then anything selected in the receiver is un-selected automatically.

The multiple receiver for frequency tables is an example.  I tabulate columns x and y in that order.  I would now like to replace x by q.  If I delete x, then y moves up, so I need to clear the whole receiver and start again!

Am I missing something here?

**Comment by dannyparsons** [comment]
No you're not missing anything, that's how it works at the moment. We should look into how we could achieve that functionality.

**Comment by dannyparsons** [comment]
Something other than a `ListView` control would be needed to achieve this functionality and I'm open to suggestions of a different control which keeps the same basic functionality we currently have.

**Comment by Patowhiz** [comment]
@dannyparsons @rdstern  what about building in a functionality to rearrange the variables (Shifting up and down) inside the multi receiver? 
So you will only  need to change the order in any way that you wish .  I think the issue here is the order in which the variables are tabulated, so after adding the new variable you just shift it to the position you want.

**Comment by dannyparsons** [comment]
Yes if we could reorder the variables then it significantly increases the need to remove and re-add variables in many cases. It doesn't quite address the original issue in the most simple way, but that would be harder to implement and I like adding reordering because its simple and solves many issues not just this one.
My concern is how to do this without taking up a lot of space on the dialog, or reducing the amount of text you see for each variable.

**Comment by Patowhiz** [comment]
We could implement this in different ways depending on how easy it is to the user; 

1. Through drag and drop
2. Through  buttons - we could implement floating buttons that become visible when an item is selected and the receiver has focus. 

Either of the above could be done in away that extra space is not needed or just a tiny space will be needed.(I think so)

I'm not sure how this will be perceived by our users.

**Comment by dannyparsons** [comment]
Drag and drop would be my choice, but I don't assume that will be simple. Maybe look into that first and see what options are, it may be simple with a different type of control?
